go/ai/prompt.go

```diff
@@ -37,18 +37,26 @@ type PromptRequest struct {
 	Model string `json:"model,omitempty"`
 }
 
-// Prompt is the interface used to execute a prompt template and
-// pass the result to a [ModelAction].
-type Prompt interface {
-	Generate(context.Context, *PromptRequest, func(context.Context, *Candidate) error) (*GenerateResponse, error)
-}
+// A PromptAction is used to generate content from an AI model using a prompt.
+type PromptAction = core.Action[*PromptRequest, *GenerateResponse, *Candidate]
```

Comment:
Maybe I'm misreading the JS, but it seems to me that the prompt action doesn't do generation, it just produces the input:
```
type PromptAction<I extends z.ZodTypeAny = z.ZodTypeAny> = Action<
   I,
    typeof GenerateRequestSchema>
```
which seems to correspond to the (currently invalid) Go
```
type PromptAction[I any] = Action[I, *GenerateRequest, struct{}]
```

Reply:
Hmmm, I expect you are right.  Now I'm not sure what `PromptAction` is for on the TypeScript side.

Reply:
Presumably you call the prompt to get the request, then call ai.Generate with the request, but I haven't checked.